core/src/main/java/com/redhat/lightblue/client/LightblueClientConfiguration.java

```diff
@@ -6,6 +6,21 @@
 
 public class LightblueClientConfiguration {
 
+    public enum Compression {
+        NONE, LZF;
+
+        public static Compression parseCompression(String str) {
+            switch (str) {
+            case "NONE":
+                return NONE;
+            case "LZF":
+                return LZF;
+            default:
+                throw new IllegalArgumentException("Invalid compression " + str + ". Supported values are: NONE, LZF");
```

Comment:
The protocol ensures backwards compatibility. If the client advertises ability to decompress responses, but lightblue doesn't set content-encoding header to lzf, then the client will treat the response as not compressed.

Lzf is considered to be very fast and I doubt it will slow things down in any scenario (even a very fast network). On slower networks it will offer better response times.

For those reasons, I made compression on by default.